### PR TITLE
Use index from status for policy-templates

### PR DESCRIPTION
**Pull request by dhaiducek**
The name from the status was used, but when a policy has policy-templates of different kinds with the same name, only the severity from the first policy-template is returned and is used for all policy-templates with that name.

ref: https://issues.redhat.com/browse/ACM-6350

### CODE REVIEW

**Comment by dhaiducek** [comment]
/cc @mprahl @gparvin 

I went the route of assuming that the index of the status was the same as the policy-templates and bailing if the name was different (which is the only data point in the status to work from), but I can restore the loop as a fallback if we think that might be necessary.

File: pkg/processor/reportprocessor.go [hunk 1]
```diff
@@ -273,69 +273,83 @@ func getGovernanceResults(dynamicClient dynamic.Interface, clusterInfo types.Man
...
-			glog.Warningf("error parsing objectDefinition as a map for policy %s, template %d: %s", plcName, idx, err)
-			continue
-		}
-		objDefName, _, err := unstructured.NestedString(objDef, "metadata", "name")
-		if err != nil {
-			glog.Warningf("error parsing objectDefinition name as a string for policy %s, template %d: %s", plcName, idx, err)
-			continue
-		}
+	// Bail if the provided index is out of range
+	if len(plcTemplates) < statusIndex {
```
Comment (mprahl) [inline pkg/processor/reportprocessor.go:286]:
```suggestion
	if len(plcTemplates) - 1 < statusIndex {
```

**Review by mprahl — APPROVED** [review]

**Review by mprahl — APPROVED** [review]
/unhold

**Comment by dhaiducek** [comment]
/cc @jlpadilla 

Remember that one time I said that would be the final PR. Well, here it is. The final final PR. 😂

**Comment by dhaiducek** [comment]
(FYI: I created an issue for us to add tests here in the future: https://issues.redhat.com/browse/ACM-7401)

**Comment by dhaiducek** [comment]
I deployed this to our dev cluster, and it works as expected. I'll go ahead and add it to the 2.8 cherry-pick.

@jlpadilla Please take a look when you have a moment.

**Pull request merged**